railties/lib/rails/railtie.rb

```diff
@@ -50,18 +50,18 @@ module Rails
   # To add an initialization step to the \Rails boot process from your railtie, just
   # define the initialization code with the +initializer+ macro:
   #
-  #   class MyRailtie < Rails::Railtie
```

Comment:
My first thought was keep `MyRailtie` because it's obvious we're defining a custom class we own, not to be confused with the `Railtie` class. It maybe a convention that Rails follows for the framework itself, but that doesn't mean every gem has to apply that same naming. And actually that is probably more confusing to a beginner reading about this from that class.

On second thought, given the path is `my_gem/railtie` it's probably _better_ to name it following Zeitwerk's naming preference and then use autoload. We can also probably remove the require in the previous example, then?

Need to give this more thought.

There is also a [guide](https://edgeguides.rubyonrails.org/plugins.html) for creating plugins, and I wonder if that is not a better place for this type of complete tutorial. I can imagine the guide can give a full in-depth example and reference back to this class for more details, once that is established the naming and assumptions in this API doc can be more consistent.

Since the Rails Foundation is revamping the guides, I have made a note to pass on this information when they get to Plugins.

Reply:
@zzak agreed that the explicit require could go away. i'll cut it out while you ponder possible next steps here.